cpp/arcticdb/storage/common.hpp

```diff
@@ -91,5 +92,17 @@ struct is_key_type<entity::VariantKey> : std::true_type {};
 template <typename T>
 inline constexpr bool is_key_type_v = is_key_type<T>::value;
 
-}  //namespace arcticdb::storage
-
+class NativeVariantStorage {
+public:
+    using VariantStorageConfig = std::variant<std::monostate, s3::S3Settings>;
+    explicit NativeVariantStorage(VariantStorageConfig config) : config_(std::move(config)) {};
```

Comment:
Might be slightly nicer to have a zero argument constructor that sets the `std::monostate` and then constructors for each other member of the variant type.

Reply:
Not a huge fan of default paramter but maybe it's a good use case here to conslidate the constructor. It should be just a very thin wrapper of the variant so can't forsee the default parameter will be a nightmare in the future